lib/core/src/Cardano/Pool/DB/Model.hs

```diff
@@ -453,6 +449,8 @@ mRemovePools poolsToRemove = do
         $ Map.filterWithKey $ \(_, p) _ -> retain p
     modify #retirements
         $ Map.filterWithKey $ \(_, p) _ -> retain p
+    modify #delisted
```

Comment:
Do we really want this? I'd say that the pool delisted set should just reflect what SMASH returns, without any local modifications.

Reply:
Also note: the next query to SMASH will add the just deleted poolIds again.

Reply:
Well, it all depends on whether you want the set of pool IDs in `pool_delistment` to always be a subset of the pools that we know about.

I'd say it's probably okay to remove them when we remove pools.

Reply:
But we're not yet satisfying that property, see https://github.com/input-output-hk/cardano-wallet/pull/2277/files#diff-7221813f41d4c796791728852d1b5e3f2007929b3c613e3d07dda5484d11d486R494-R496

Reply:
> Do we really want this? I'd say that the pool delisted set should just reflect what SMASH returns, without any local modifications?

Fair enough. I've fixed this in 12f5426917b675d1c9ee01fc2f48510d95ecf422.